data/src/main/java/com/linkedin/data/schema/validation/ValidateDataAgainstSchema.java

```diff
@@ -505,11 +505,11 @@ else if (object instanceof DataMap)
       {
         // Pegasus mode
         DataMap map = (DataMap) object;
-        if (map.size() != 1)
+        if (map.size() > 1)
```

Comment:
Do you mind adding a comment here that we allow empty union

Reply:
Added some comment